tools/ci_build/provider_ops_disabler.py

```diff
@@ -0,0 +1,380 @@
+            call_back_ret = call_back(op_type, opset, opset, domain)
+
+        elif onnx_typed_op in code_line:
+            # e.g. class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 8, float, Expand);
+            # e.g. BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 7, double, Sin)>,
+            trim_at = code_line.index(onnx_typed_op) + onnx_typed_op_len
+            *_, domain, opset, _, op_type =\
+                [arg.strip() for arg in code_line[trim_at: -len(end_mark)].split(',')]
+            call_back_ret = call_back(op_type, opset, opset, domain)
+
+        elif onnx_versioned_op in code_line:
+            # e.g. class ONNX_OPERATOR_VERSIONED_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 1, 10, Unsqueeze);
+            # e.g. BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 1, 10, Hardmax)>,
+            trim_at = code_line.index(onnx_versioned_op) + onnx_versioned_op_len
+            *_, domain, opset_from, opset_to, op_type =\
+                [arg.strip() for arg in code_line[trim_at: -len(end_mark)].split(',')]
+            call_back_ret = call_back(op_type, opset_from, opset_to, domain)
+
+        elif onnx_versioned_typed_op in code_line:
+            # e.g. class ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 7, 9, float, Upsample);
+            # e.g. BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 1, 10, float, LogSoftmax)>,
+            trim_at = code_line.index(onnx_versioned_typed_op) + onnx_versioned_typed_op_len
+            *_, domain, opset_from, opset_to, _, op_type =\
+                [arg.strip() for arg in code_line[trim_at: -len(end_mark)].split(',')]
+            call_back_ret = call_back(op_type, opset_from, opset_to, domain)
+
+        return offset + 1, call_back_ret  # end of process_lines(...)
+
+    lines = []
+    with open(ep_path, 'r') as file_to_read:
+        lines = file_to_read.readlines()
+
+    backup_path = ep_path + '~'
+    if not os.path.isfile(backup_path):
+        shutil.move(ep_path, backup_path)
+
+    with open(ep_path, 'w') as file_to_write:
+        line_offset = 0
+
+        while line_offset < len(lines):
+
+            line = lines[line_offset]
+            stripped = line.strip()
+
+            if stripped.startswith('class ONNX_OPERATOR'):
+                # collect op versions
+
+                next_line_offset, _ = process_lines(lines,
+                                                    line_offset,
+                                                    tuple([');']),
+                                                    fill_version_map)
+
```

Comment:
The original script commented out the forward declarations as well as the registrations, so for each line in either it checked the version info vs the required ops and included or excluded the line.

What's the purpose of creating a version map from the forward declarations (and not commenting them out) and using that when excluding the registrations? If the forward declaration doesn't match the registration the registration will break. If we don't need to comment out the forward declarations to achieve the binary size reduction couldn't we ignore those lines completely, and just check version info against required ops as we process each line instead of building a map first?